types/lodash/index.d.ts

```diff
@@ -8,7 +8,7 @@
 //                 Junyoung Clare Jang <https://github.com/ailrun>,
 //                 e-cloud <https://github.com/e-cloud>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.4
```

Comment:
Why is typescript 2.4 required? Aren't we still compatible with typescript 2.2?

Reply:
@aj-r you could try to fork the PR and change it back and see what happen. With the latest dtslint, lodash type is tested across various versions of typescript. And it throws error about the `lodash-test.ts` saying something is not compatible with TS v2.3 but TS v2.4. Therefore, the simplest way is to update the target version.

May be there is another way to provide more proper tests compatible with typescript v2.2. But for now i can't figure it out.

Reply:
@aj-r you could try to fork the PR and change it back and see what happen. With the latest dtslint, lodash type is tested across various versions of typescript. And it throws error about the `lodash-test.ts` saying something is not compatible with TS v2.3 but TS v2.4. Therefore, the simplest way is to update the target version.

May be there is another way to provide more proper tests compatible with typescript v2.2. But for now i can't figure it out.

Reply:
@aj-r you could try to fork the PR and change it back and see what happen. With the latest dtslint, lodash type is tested across various versions of typescript. And it throws error about the `lodash-test.ts` saying something is not compatible with TS v2.3 but TS v2.4. Therefore, the simplest way is to update the target version.

May be there is another way to provide more proper tests compatible with typescript v2.2. But for now i can't figure it out.

Reply:
@aj-r you could try to fork the PR and change it back and see what happen. With the latest dtslint, lodash type is tested across various versions of typescript. And it throws error about the `lodash-test.ts` saying something is not compatible with TS v2.3 but TS v2.4. Therefore, the simplest way is to update the target version.

May be there is another way to provide more proper tests compatible with typescript v2.2. But for now i can't figure it out.

Reply:
@aj-r you could try to fork the PR and change it back and see what happen. With the latest dtslint, lodash type is tested across various versions of typescript. And it throws error about the `lodash-test.ts` saying something is not compatible with TS v2.3 but TS v2.4. Therefore, the simplest way is to update the target version.

May be there is another way to provide more proper tests compatible with typescript v2.2. But for now i can't figure it out.

Reply:
@e-cloud I see - it looks like the error is only in `lodash-tests.ts`, not in `index.d.ts`, so I think the correct thing is to fix `lodash-tests`, not change the typescript version.  I think the problem is that type inference isn't working correctly for `ListOfRecursiveArraysOrValues`.

```
Type 'LoDashImplicitArrayWrapper<{}>' is not assignable to type 'LoDashImplicitArrayWrapper<number>'.
  Type '{}' is not assignable to type 'number'.
```

I also see a few other small issues so I'll fix them when I get time.